v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+        metadata += sizeof("func") + 1;
+
+        parsed_events[current_parsed_event_count].line = *(int32_t*)current_user_data_field;
+        current_user_data_field+= sizeof(int32_t);
+
+        POOR_MANS_ASSERT(strcmp((const char*)metadata, "line") == 0);
+        metadata += sizeof("line") + 1;
+
+        uint32_t property_index = 0;
+
+        // get all other properties
+        while (current_user_data_field - (const char*)pEventRecord->UserData < pEventRecord->UserDataLength)
+        {
+            // get name and type from metadata
+            (void)strncpy(parsed_events[current_parsed_event_count].properties[property_index].property_name, (const char*)metadata, sizeof(parsed_events[current_parsed_event_count].properties[property_index].property_name));
+            parsed_events[current_parsed_event_count].properties[property_index].property_name[sizeof(parsed_events[current_parsed_event_count].properties[property_index].property_name) - 1] = '\0';
+            metadata += strlen(parsed_events[current_parsed_event_count].properties[property_index].property_name) + 1;
+            parsed_events[current_parsed_event_count].properties[property_index].property_type = *metadata;
+            metadata++;
+
+            switch (parsed_events[current_parsed_event_count].properties[property_index].property_type)
+            {
+            default:
+                POOR_MANS_ASSERT(0);
+                break;
+            case _TlgInSTRUCT | _TlgInChain:
+                parsed_events[current_parsed_event_count].properties[property_index].struct_field_count = *(uint8_t*)metadata;
+                metadata++;
+                break;
+            case TlgInANSISTRING:
+                (void)strncpy(parsed_events[current_parsed_event_count].properties[property_index].ascii_char_ptr_value, current_user_data_field, sizeof(parsed_events[current_parsed_event_count].properties[property_index].ascii_char_ptr_value));
+                parsed_events[current_parsed_event_count].properties[property_index].ascii_char_ptr_value[sizeof(parsed_events[current_parsed_event_count].properties[property_index].ascii_char_ptr_value) - 1] = '\0';
+                current_user_data_field += strlen(current_user_data_field) + 1;
+                break;
+            case TlgInUINT8:
+                parsed_events[current_parsed_event_count].properties[property_index].uint8_t_value = *(uint8_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint8_t);
+                break;
+            case TlgInINT8:
+                parsed_events[current_parsed_event_count].properties[property_index].int8_t_value = *(int8_t*)current_user_data_field;
+                current_user_data_field += sizeof(int8_t);
+                break;
+            case TlgInUINT16:
+                parsed_events[current_parsed_event_count].properties[property_index].uint16_t_value = *(uint16_t*)current_user_data_field;
```

Comment:
>uint16_t [](http://example.com/codeflow?start=104&length=8)

I don't think this is right.

`current_user_data_field` is not guaranteed to be aligned (at all). Casting and reading unaligned works wonder on Intel platforms, on others... less so.

Let's use some primitives that read byte by byte and assemble the bytes in "values" after the bytes are read.

Also, we can only hope that the endianess of the bytes matches what we expect as a platform (different concern).

Reply:
Added some local read functions for this